### DISCUSSION TITLE
Missed expand button for custom subfields - The Scandinavian

### DISCUSSION THREAD

**Discussion by OmarHeshamMohamed** [microsoft/Dynamics365Commerce.Solutions#32] [Q&A]
Recently cx faced issue related to incorrect display of custom cart line subfield on cart page.
they have custom subfield that display data from cart line extension properties and operation that update this property and call standard refresh cart operation to trigger UI update. It works fine if select one of existing line to add subfield property, but if you add new cart line and then trying to add value for subfield to it, POS for some reason doesn’t trigger UI logic to display new subfield although refresh cart operation receives updated cart with all new extension properties that should be displayed.

 
The issue is that the last cart line that have session property updated doesn't display it as subfield.

1-	add one product to the cart and then set new extension property to it.
![image](https://user-images.githubusercontent.com/107184611/220688675-8c4c3e2c-fa96-479f-99be-fed9d6b01a4b.png)

2- add another 2 lines but properties not displayed

![image](https://user-images.githubusercontent.com/107184611/220688895-7f1d5969-f0d3-461c-8985-b2aed3117b44.png)

3- 	if we tried to modify the second line we can see extension property will be displayed

![image](https://user-images.githubusercontent.com/107184611/220689195-071e2cd1-b5f2-4200-823f-8c01b94fafad.png)

 
![image](https://user-images.githubusercontent.com/107184611/220685309-c2bd3dab-1fa9-4340-9a25-2a67278dab07.png)

**Comment by madyke** [comment]
Hi @OmarHeshamMohamed - there are couple of things that could be going on here. There are some optimizations that are in place for the cart page to prevent re-rendering of unchanged parts of the page when only some cart lines are updated. It's possible that could be affecting this, or there could be a bug that is preventing new lines from being updated.

For us to take a deeper look and diagnose the issue, can you please provide a link to a sample where this issue reproduces. You can create a fork of a sample repo and modify it to reproduce the issue. If you need help, please see the link below on how to fork a GitHub repo or refer to the repo Wiki on how to create a minimal repro.
https://docs.github.com/en/get-started/quickstart/fork-a-repo

Also please provide specific steps to reproduce the behavior, such as:
1. Go to '...'
2. Click on '....'
3. Run command '....'
4. See error
5. etc.

**Reply by AndersXNielsen** [reply]
Hi.

Here is a video demonstrating the problem :

[https://1drv.ms/v/s!Ah1gID91c75IhmDBMLdmskGE6s6T?e=f21eLP](url)

**Reply by ahgamal-ms** [reply]
@madyke we have same issue reported for another customer , they have an issue related to incorrect display of custom cart line subfield on cart page.
We have custom subfield that display data from cart line extension properties and operation that update this property and call standard refresh cart operation to trigger UI update. It works fine if select one of existing line to add subfield property, but if you add new cart line and then trying to add value for subfield to it, POS for some reason doesn’t trigger UI logic to display new subfield although refresh cart operation receives updated cart with all new extension properties that should be displayed.

**Reply by madyke** [reply]
We have filed an internal bug to review this, unfortunately due to other high-priority holiday feature commitments, we may not get to it for a while.
